### ISSUE TITLE
CI/CD Integration for Automated Release Builds and Notifications

### ISSUE THREAD

**Issue by esmaeil-ahmadipour** [pactus-project/pactus-gui#47]
#### Feature Description  
Set up CI/CD integration to automatically create release builds for Windows, Linux, and macOS after any successful merge request to the `develop` branch. The builds should then be shared on both Telegram and Discord channels. This will streamline the release process and ensure timely notifications.

<details>
<summary>Click to expand the scenario</summary>

### Scenario: Automated Release Sharing  

When a developer merges a feature branch into the `develop` branch, the CI/CD pipeline automatically generates release builds for Windows, Linux, and macOS. These builds are then shared in a designated Telegram group and a Discord channel, ensuring the team is informed and has access to the latest versions.

</details>

---

### API Integration  
- **Telegram Bot API Endpoint:**  
  - **Endpoint:** `https://api.telegram.org/bot{BOT_TOKEN}/sendMessage`  
  - **Required Fields:** `chat_id`, `text`, and optionally `file` for attaching builds.  

- **Discord Webhook Endpoint:**  
  - **Endpoint:** `{DISCORD_WEBHOOK_URL}`  
  - **Required Data:** `content` for the message and optional file attachments.  
  
  ---
  
 ### Additional Notes:
- **Merge Request Target:** This feature should only be triggered after a successful merge request to the `develop` branch.
- **Build Section:** If possible, split the build section into a separate file for better maintainability.

---

**UPDATE**

Ways to access and download the latest release, upon acceptance of the PR, are listed below:

- Github Releases : [Link (Releases)](https://github.com/pactus-project/pactus-gui/releases)
- Telegram Cahnnel : [Link (PactusGuiRelease)](https://t.me/pactusgui)
- Discord Cahnnel : [Link (PactusGuiRelease)](https://discord.com/channels/1324099313726390365/1324099314456465523)

**Comment by phoenixit99** [comment]
📊 Current Status:

Progress: 30%
✔️ Completed Tasks:

Sent message to Integram:
Successfully communicated project updates to our Telegram channel, ensuring all team members are informed of our current progress and upcoming tasks.
🔄 Pending Tasks:

Send message to Discord:
We need to update our Discord channel with the latest project developments. This will ensure that all stakeholders have access to the same information and can provide feedback or ask questions as needed.


🎉 Pull Request Merged!

PR Title: The title of the pull request
Merged by: The author's name
Repository: The repository name
Branch: dev/test
[View Pull Request](the link of the pull request)

Changes Summary:

**Comment by phoenixit99** [comment]
#### Release page links   
![Screenshot 1](https://raw.githubusercontent.com/phoenixit99/flutter_clean/main/lib/assets/Screenshot%202024-12-31%20at%203.43.57%E2%80%AFPM.png)

#### Integrated Discord notifications
![Screenshot 1](https://raw.githubusercontent.com/phoenixit99/flutter_clean/main/lib/assets/Screenshot%202024-12-31%20at%204.09.18%E2%80%AFPM.png)

#### Integrated Telegram notifications
![Screenshot 1](https://raw.githubusercontent.com/phoenixit99/flutter_clean/main/lib/assets/Screenshot%202024-12-31%20at%204.10.09%E2%80%AFPM.png)
 
#### Github workflow build - release - notification 
![Screenshot 1](https://raw.githubusercontent.com/phoenixit99/flutter_clean/main/lib/assets/Screenshot%202024-12-31%20at%204.09.57%E2%80%AFPM.png)

**Comment by esmaeil-ahmadipour** [comment]
@phoenixit99 Great work, thanks for your efforts! I need the following two points to be added to your task:

1. **Standardize Release Names and Tags**:
   - Tag names should follow a consistent format, for example: `v1.6.4+27`.
   - Release names should follow this format: `Release 1.6.4+27`.

2. **Output for Telegram and Discord**:  
   We need outputs for Telegram and Discord so team members can easily access the app for installation. To keep things organized, it's better to generate tags and releases only for PRs related to merging `develop` into `main`.

**Comment by phoenixit99** [comment]
> @phoenixit99 Great work, thanks for your efforts! I need the following two points to be added to your task:
> 
> 1. **Standardize Release Names and Tags**:
>    
>    * Tag names should follow a consistent format, for example: `v1.6.4+27`.
>    * Release names should follow this format: `Release 1.6.4+27`.
> 2. **Output for Telegram and Discord**:
>    We need outputs for Telegram and Discord so team members can easily access the app for installation. To keep things organized, it's better to generate tags and releases only for PRs related to merging `develop` into `main`.

I agree with the workflow for the main branch.

Regarding the develop branch, I have a suggestion: we should send notifications to Telegram and Discord when there are any pull requests. This will increase team activity and help us keep track of progress.

Additionally, we should also send notifications for the daily build releases.

cc @b00f

**Comment by esmaeil-ahmadipour** [comment]
> Regarding the develop branch, I have a suggestion: we should send notifications to Telegram and Discord when there are any pull requests. This will increase team activity and help us keep track of progress.
> 
> Additionally, we should also send notifications for the daily build releases.
> 
> cc @b00f

---

I suggest creating a new issue.

### Reason for the new issue:
The CI/CD setup needs to be updated so that release builds and tags are applied only to the `main` branch. Additionally, the naming convention for tags and releases must follow the format outlined below:

### Naming Convention:
- **Tag**: v0.0.1+1
- **Release**: release-<OS>-v0.0.1+1

### Tasks:
1. Update the pipeline to target the `main` branch exclusively.
2. Ensure that tagging and release creation occur only after successful merges into the `main` branch.
3. Implement the specified naming convention for both tags and releases.
4. Verify that notifications for these releases are sent to Telegram and Discord channels.
